### PR TITLE
Adding template message documentation

### PR DESCRIPTION
**Pull request by jupierce**
Documentation for openshift/origin#9708

### CODE REVIEW

**Comment by jupierce** [comment]
@bparees

**Comment by bparees** [comment]
lgtm.
@openshift/team-documentation ptal

File: dev_guide/templates.adoc [hunk 1]
```diff
@@ -307,14 +307,21 @@ apiVersion: "v1"
...
 <1> The name of the template as it will appear to users.
-<2> A description of the template.
-<3> Tags to be associated with the template for searching and grouping.
-<4> An icon to be displayed with your template in the web console.
+<2> A user-friendly name which can be employed by user interfaces.
+<3> A description of the template.
+<4> Tags to be associated with the template for searching and grouping.
+<5> An icon to be displayed with your template in the web console.
+<6> An instructional message that will be displayed when this template is instantiated.
+This field should inform the user how to utilize the newly created resources.
```
Comment (ahardin-rh) [inline dev_guide/templates.adoc]: s/utilize/use

File: dev_guide/templates.adoc [hunk 2]
```diff
@@ -307,14 +307,21 @@ apiVersion: "v1"
...
 ----
 <1> The name of the template as it will appear to users.
-<2> A description of the template.
-<3> Tags to be associated with the template for searching and grouping.
-<4> An icon to be displayed with your template in the web console.
+<2> A user-friendly name which can be employed by user interfaces.
+<3> A description of the template.
+<4> Tags to be associated with the template for searching and grouping.
+<5> An icon to be displayed with your template in the web console.
+<6> An instructional message that will be displayed when this template is instantiated.
```
Comment (ahardin-rh) [inline dev_guide/templates.adoc]: s/will be/is

File: dev_guide/templates.adoc [hunk 3]
```diff
@@ -307,14 +307,21 @@ apiVersion: "v1"
...
+    description: "An example CakePHP application with a MySQL database" <3>
+    tags: "instant-app,php,cakephp,mysql" <4>
+    iconClass: "icon-php" <5>
+message: "Your admin credentials are ${ADMIN_USERNAME}:${ADMIN_PASSWORD}" <6>
 ----
 <1> The name of the template as it will appear to users.
-<2> A description of the template.
-<3> Tags to be associated with the template for searching and grouping.
-<4> An icon to be displayed with your template in the web console.
+<2> A user-friendly name which can be employed by user interfaces.
```
Comment (ahardin-rh) [inline dev_guide/templates.adoc]: comma after name

File: dev_guide/templates.adoc [hunk 4]
```diff
@@ -307,14 +307,21 @@ apiVersion: "v1"
...
+<3> A description of the template.
+<4> Tags to be associated with the template for searching and grouping.
+<5> An icon to be displayed with your template in the web console.
+<6> An instructional message that will be displayed when this template is instantiated.
+This field should inform the user how to utilize the newly created resources.
+Parameter substitution will be performed on the message before being displayed so
```
Comment (ahardin-rh) [inline dev_guide/templates.adoc]: s/will be/is

File: dev_guide/templates.adoc [hunk 5]
```diff
@@ -391,6 +399,8 @@ xref:../architecture/core_concepts/templates.adoc#parameters[regular
 expression-like syntax].
 <7> The input to the generator. In this case, the generator will produce a 40
 character alphanumeric value including upper and lowercase characters.
+<8> Parameters can be included in the template message. This is particularly
+useful to inform the user about generated values.
```
Comment (ahardin-rh) [inline dev_guide/templates.adoc]: s/is particularly useful to inform/informs

**Comment by ahardin-rh** [comment]
@jupierce @bparees just a few comments from me. Assuming this is for 3.3? Please confirm that my labes/milestone are correct. 

Also, looks like Travis is currently throwing ruby_dep (1.4.0) errors across all new doc contributions. We're looking into it. Thanks!

**Comment by bparees** [comment]
yes 3.3 origin/online/dedicated/enterprise

**Comment by jupierce** [comment]
Updated. Thanks @ahardin-rh .

**Comment by jupierce** [comment]
[test]

**Comment by ahardin-rh** [comment]
Carried this over to https://github.com/openshift/openshift-docs/pull/2871

**Pull request closed without merging**